### ISSUE TITLE
[XLA:CPU] reduce_window on the YNNPACK path ignores base_dilation when window size and stride are 1

### ISSUE THREAD

**Issue by EylonKrause** [openxla/xla#48714]
`reduce_window` with `base_dilation > 1` on a dimension whose window size and stride are both
1 silently drops the dilation holes when the op is lowered to the YNNPACK fusion path (large enough
inputs), in both eager and jit. The same op on a small input, which uses XLA's own emitter, is
correct.

### Reproducer

```python
import os
os.environ["JAX_PLATFORMS"] = "cpu"
import numpy as np, jax, jax.numpy as jnp
from jax import lax


def oracle_maxpool(x, wd, st, bd):
    xd = np.full(tuple((n - 1) * d + 1 for n, d in zip(x.shape, bd)), -np.inf)   # base-dilated, holes = init
    xd[tuple(slice(None, None, d) for d in bd)] = x
    out_shape = tuple((p - w) // s + 1 for p, w, s in zip(xd.shape, wd, st))
    out = np.full(out_shape, -np.inf)
    for i in range(wd[0]):
        for j in range(wd[1]):
            out = np.maximum(out, xd[i:i + (out_shape[0] - 1) * st[0] + 1:st[0], j:j + (out_shape[1] - 1) * st[1] + 1:st[1]])
    return out


wd, st, bd = (2, 1), (2, 1), (1, 2)          # dim 1: window 1, stride 1, base_dilation 2
f = lambda a: lax.reduce_window(a, -jnp.inf, lax.max, wd, st, ((0, 0), (0, 0)), base_dilation=bd)
for shape in ((8, 8), (64, 128)):            # 64 elements -> XLA emitter;  8192 elements -> ynn_fusion
    x = np.random.default_rng(0).integers(0, 100, size=shape).astype(np.float32)
    exp = oracle_maxpool(x, wd, st, bd)      # columns alternate: value, -inf, value, -inf, ...
    eager = np.asarray(f(jnp.asarray(x))); jitted = np.asarray(jax.jit(f)(jnp.asarray(x)))
    hlo = jax.jit(f).lower(jnp.asarray(x)).compile().as_text()
    print(f"shape {shape} -> out {exp.shape}, ynn_fusion={'ynn_fusion' in hlo}")
    print("  expected row0[:6]:", exp[0, :6].tolist())
    print("  eager    row0[:6]:", eager[0, :6].tolist(), "WRONG" if not np.array_equal(eager, exp) else "ok")
    print("  jit      row0[:6]:", jitted[0, :6].tolist(), "WRONG" if not np.array_equal(jitted, exp) else "ok")
```

Output on my machine:

```
shape (8, 8) -> out (4, 15), ynn_fusion=False
  expected row0[:6]: [85.0, -inf, 81.0, -inf, 64.0, -inf]
  eager    row0[:6]: [85.0, -inf, 81.0, -inf, 64.0, -inf] ok
  jit      row0[:6]: [85.0, -inf, 81.0, -inf, 64.0, -inf] ok
shape (64, 128) -> out (32, 255), ynn_fusion=True
  expected row0[:6]: [89.0, -inf, 63.0, -inf, 90.0, -inf]
  eager    row0[:6]: [89.0, 63.0, 90.0, 67.0, 89.0, 19.0] WRONG
  jit      row0[:6]: [89.0, 63.0, 90.0, 67.0, 89.0, 19.0] WRONG
```

### Analysis

Base dilation inserts `base_dilation - 1` copies of the init value between elements, so every odd
column of the result must be `-inf` here. On the YNNPACK path the holes are gone and the output is
the undilated data.

The dilation checks in `IsReduceLikeOpSupportedByYnn` (`xla/backends/cpu/ynn_support.cc`) sit inside
a `if (dim.size() > 1 || dim.stride() > 1)` block, so a dimension with window size 1 and stride 1 is
never checked for `base_dilation != 1`; `DefineReduceWindowOp`
(`xla/backends/cpu/ynn_emitter.cc`) then emits a static pad plus stencil copy with no dilation. The
output shape is computed correctly, which is why the mismatch is silent.

This is reachable from ordinary code: JAX's `reduce_window` transpose rule sets
`base_dilation = strides`, so the gradient of a strided pooling window of size 1 lands here.

### Environment

- `jax` 0.11.1, `jaxlib` 0.11.1 (pip), CPU backend, `JAX_PLATFORMS=cpu`
- Linux x86-64 (AVX-512), Python 3.12

### Notes

Found by a differential fuzzer comparing XLA:CPU against a NumPy oracle written from the operation semantics (validated by hand and against `jax._src.lax_reference`), then minimised. I am filing this as an issue rather than a pull request, per the repository's contribution guidelines: this report was prepared with the help of an AI coding assistant (Claude), and I reproduced every result above myself on the environment listed.

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/982248574.
